modules/debugging/index.qmd

```diff
@@ -91,33 +99,281 @@ detach_package(tidyverse)
+|                                                                           | -   Don't be disheartened if your error, which you've struggled with for hours, is solved in minutes — we've had plenty of practice (also being stuck for hours) troubleshooting these errors.                                                                                                                                                                                                                                 |
++---------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+
+::: callout-tip
+## Help and Learn
+
+Answering questions on the forum can help consolidate your understanding and prepare you for asking and answering questions in other forums such as Stack Overflow.
+:::
+
+# Minimal Reproducible Examples
+
+::: callout-note
+## Remember
+
+A minimally reproducible example (often called an MRE) contains all of the code necessary for someone else to encounter the same error you're facing.
+:::
+
+To create one, you need to provide the **exact packages you're using**, some **data** if needed, and **your code.** A good example is also **minimal**, which means that you've taken the time to remove as much unrelated packages, code and data as possible to produce the error.
+
+## Providing minimal datasets
+
+If your question involves using a dataset, you will need to provide a small dataset to demonstrate the problem. Often people say that the data is confidential and can’t be provided, but usually the problem isn’t specific to the data you are using. Taking the time to try your code on a different dataset is a good debugging step, and R provides many good datasets to use in these examples built in. You can find a complete list of datasets available to all R users with the `data(package = "datasets")` function.
+
+If you do need to use a custom dataset specific to your project, try reducing the size of the dataset before making your example. The error may only occur for some specific rows of your data. Once you've narrowed in on the issue, you can convert your minimal data object to code for producing it using the `dput()` function.
+
+For example you can use use the below code to provide the first 6 rows of the iris dataset.
+
+```{r}
+dput(head(iris))
```

Comment:
Better use a different dataset, like penguins (iris has history in eugenics).
penguins is also nicer as it allows sharing data without requiring the palmerpenguins package to be installed.

edit: I see you use penguins as an exercise, great! mtcars or similar would be a better alternative.